R/use_lintr.R

```diff
@@ -43,5 +43,33 @@ use_lintr <- function(path = ".", type = c("tidyverse", "full")) {
     )
   )
   write.dcf(the_config, config_file, width = Inf)
+
+  # If this is an R package and if available, add .lintr file to .Rbuildignore
+  if (file.exists("DESCRIPTION") && file.exists(".Rbuildignore")) {
+    try(
+      add_build_ignore(config_file),
+      silent = TRUE
+    )
+  }
+
   invisible(config_file)
 }
+
+add_build_ignore <- function(path) {
+  path_build_ignore <- ".Rbuildignore"
+  existing_lines <- readLines(path_build_ignore, encoding = "UTF-8", warn = FALSE)
+
+  escape_path <- gsub("^\\./", "", path)
+  escape_path <- gsub("\\.", "\\\\.", escape_path)
+  escape_path <- gsub("/$", "", escape_path)
+  escape_path <- paste0("^", escape_path, "$")
```

Comment:
This could be `rex::rex(start, rex::re_substitutes(path, list(start, "./") %or% list("/", end), ""), end)` and it would safely escape any chars in path.